### PR TITLE
Split tjpgdec into own crate

### LINKED ISSUE #2787: Separate JPEG decoder into its own crate
Let's call it `trezor-tjpgdec` maybe?

cc @TychoVrahe

### PR DESCRIPTION
**Pull request by mmilata**
I put the crate under `rust/trezor-tjpgdec` because at the time slack discussion suggested this is going to be where we keep rust crates. Putting this into separate repo is certainly an option.

No functional changes except replacing `Rect`/`Point`/`Offset` with `(u32, u32)`.

Not sure what to do with the license, the original library has [custom bsd-like one](http://elm-chan.org/fsw/tjpgd/en/appnote.html#license) while the cmumford fork has GPL3 which i used for the time being.

Fixes #2787.

### CODE REVIEW

File: rust/trezor-tjpgdec/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# trezor-tjpgdec
+
+Rust version of [Tiny JPEG Decompressor](http://elm-chan.org/fsw/tjpgd/00index.html).
+
+## Credits
+
+* Original C library was written by [ChaN](http://elm-chan.org/).
+* Several bugs fixed by [Chris Mumford](https://github.com/cmumford/TJpgDec).
+* Converted to Rust using [c2rust](https://c2rust.com/) by [Martin Milata](mailto:martin.milata@satoshilabs.com).
+* Heavy refactoring by [Michal Bruna](mailto:michal.brun@satoshilabs.com).
```
Comment (grdddj) [inline rust/trezor-tjpgdec/README.md:10]: Missing `a` from Michal's surname in email :)

Comment (mmilata) [reply rust/trezor-tjpgdec/README.md]: omg, fixed

**Review by TychoVrahe — APPROVED** [review]
LGTM. Not sure about the licence either though.

(in time, we can take this a bit further by moving into separate repo and publishing on crates.io, configuration will need to be enabled somehow for this to be useful)

**Comment by matejcik** [comment]
License is ok. the original page explicitly states:
>When redistribute TJpgDec with any modification, the license can also be changed to GNU GPL or any BSD-style license.

**Comment by matejcik** [comment]
@mmilata please merge this if you're done with it

**Pull request merged**